core/src/main/scala/kafka/raft/KafkaNetworkChannel.scala

```diff
@@ -216,11 +216,9 @@ class KafkaNetworkChannel(time: Time,
     endpoints.put(id, node)
   }
 
-  def postInboundRequest(header: RequestHeader,
-                         request: AbstractRequest,
-                         onResponseReceived: ResponseHandler): Unit = {
+  def postInboundRequest(request: AbstractRequest, onResponseReceived: ResponseHandler): Unit = {
```

Comment:
nit: maybe add a comment indicating this function is for testing only (and hence we do not care about the correlation id).

Reply:
This was actually a bug. The implementation was treating `correlationId` as unique across all connections, which of course was wrong. My fix was just to overwrite the `correlationId` from the header with one that could be unique, but obviously this loses traceability through the Raft layer. If it's ok with you, I'd like to address this problem more generally in a follow-up.

Reply:
> Unrelated with this change but when reading the code. I was confused by the name pollInboundResponses. That function returns both pending request and responses.

I don't think that's right. It only delivers responses for outbound requests (in other words, inbound responses).

In regard to the confusion, I took an unconventional turn in the beginning by consolidating inbound and outbound requests through a single API. This was useful because it allowed inputs and outputs to be serialized, which made it easier to have deterministic simulation tests. The main defect from the perspective of Raft is that fetch requests cannot be handled concurrently. I think we are going to have to address this problem in the long term, but it is outside the scope of this PR.

Reply:
> I don't think that's right. It only delivers responses for outbound requests (in other words, inbound responses).

Hmm. I misspoke a bit. I should have said that `pollInboundResponses` returns both inbound requests and responses. These are my observations:

`postInboundRequest` puts requests in `undelivered`:
https://github.com/apache/kafka/pull/9418/files#diff-c9b2a05e69be1d2391d91a33f408eab4fde1245d4ea9cf9f52305c8cb89208ceR224

`pollInboundResponses` returns everything in `undelivered`:
https://github.com/apache/kafka/pull/9418/files#diff-c9b2a05e69be1d2391d91a33f408eab4fde1245d4ea9cf9f52305c8cb89208ceR201-R202

Reply:
My mistake. This _was_ being used for both inbound requests and responses. Let me change the name and add some docs.